radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/PeerChannel.java

```diff
@@ -259,15 +259,41 @@ private void handleMessage(ByteBuf buf) throws IOException {
   @Override
   public void channelActive(ChannelHandlerContext ctx) {
     // if we weren't able to determine peer's address earlier, it should be available now
+    var peerAddress = this.nettyChannel.remoteAddress();
+
     if (this.remoteAddress.isEmpty()) {
-      this.remoteAddress = Optional.ofNullable(this.nettyChannel.remoteAddress());
+      this.remoteAddress = Optional.ofNullable(peerAddress);
+    }
+
+    if (peerAddress != null) {
```

Comment:
What if it's null? I think we might want to close the channel in that case anyway.
Something like this perhaps?:
```
if (peerAddress == null) {
  log("invalid address");
  this.disconnect();
} else if (!isSubnetAllowed(peerAddress)) {
  log("not allowed");
  this.disconnect();
} else if (this.state == ChannelState.INACTIVE) {
  this.init()
}
```

with that, we could drop the extra `REJECTED` state.

Reply:
OK, makes sense